src/db.rs

```diff
@@ -38,315 +39,338 @@ impl Default for Connection {
-    let mut dbconn = get_dbconn()?;
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    let con_row = sqlx::query(query.as_str()).fetch_one(&dbpool).await?;
 
-    let mqstr = "SELECT allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids FROM policy WHERE id = ? AND valid = 1";
-    let polval: Vec<(String, String, u32, u32, String)> = trnsx.exec(mqstr, (pid,))?;
-    Ok(policy::Policy {
-        allowed_digests: serde_json::from_str(&polval[0].0).unwrap(),
-        allowed_policies: serde_json::from_str(&polval[0].1).unwrap(),
-        min_fw_api_major: polval[0].2,
-        min_fw_api_minor: polval[0].3,
-        allowed_build_ids: serde_json::from_str(&polval[0].4).unwrap(),
+    Ok(Connection {
+        policy: con_row.try_get::<i32, _>(0)? as u32,
+        fw_api_major: con_row.try_get::<i32, _>(1)? as u32,
+        fw_api_minor: con_row.try_get::<i32, _>(2)? as u32,
+        fw_build_id: con_row.try_get::<i32, _>(3)? as u32,
+        launch_description: con_row.try_get::<String, _>(4)?,
+        fw_digest: con_row.try_get::<String, _>(5)?,
     })
 }
 
-pub fn delete_policy(pid: &u64) -> Result<()> {
-    let mut dbconn = get_dbconn()?;
+pub async fn delete_connection(uuid: Uuid) -> Result<Uuid> {
+    let dbpool = get_dbpool().await?;
 
-    let mqstr = "DELETE from policy WHERE id = ?";
+    let query = format!(
+        "DELETE from conn_bundle WHERE id = \"{}\"",
+        uuid.as_hyphenated()
+    );
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-    trnsx.exec_drop(mqstr, (&pid,))?;
-    trnsx.commit()?;
-    Ok(())
+    sqlx::query(query.as_str()).execute(&dbpool).await?;
+    Ok(uuid)
 }
 
-pub fn delete_connection(uuid: Uuid) -> Result<Uuid> {
-    let mut dbconn = get_dbconn()?;
+pub async fn insert_policy(policy: &policy::Policy) -> Result<u64> {
+    let dbpool = get_dbpool().await?;
 
-    let mqstr = "DELETE from conn_bundle WHERE id = ?";
+    //let allowed_digests_json = serde_json::to_string(&policy.allowed_digests)?;
+    //let allowed_policy_json = serde_json::to_string(&policy.allowed_policies)?;
+    //let allowed_build_ids_json = serde_json::to_string(&policy.allowed_build_ids)?;
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-    trnsx.exec_drop(mqstr, (uuid.as_hyphenated().to_string(),))?;
-    trnsx.commit()?;
-    Ok(uuid)
+    let query = format!(
+        "INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids, create_date, valid) VALUES(\'{:?}\', \'{:?}\', {}, {}, \'{:?}\', NOW(), 1)",
+        policy.allowed_digests,
+        policy.allowed_policies,
+        policy.min_fw_api_major,
+        policy.min_fw_api_minor,
+        policy.allowed_build_ids
+    );
+
+    let last_insert_row = sqlx::query(query.as_str())
+        .execute(&dbpool)
+        .await?
+        .last_insert_id();
+
+    let last_insert_id = last_insert_row.unwrap();
```

Comment:
unwrap will panic if the Result is an Err. Instead try:

```rust
let last_insert_id = last_insert_row?;
```

(untested)

Reply:
Using a ? won't work here because last_insert_id() returns an option.  I replaced unwrap with ok_or_else(|| anyhow!("db::insert_policy - error, unable to get last_insert_row"))?;  This will return an error that we can use elsewhere if we can't get the last_insert_id.